CI random failure where using `working-directory`

CI randomly failure where we are using `working-directory` key with this error
```
+ sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
cp: cannot stat '/home/runner/work/rook': No such file or directory
Error: Process completed with exit code 1.
```

seems like this field is not required anymore.

action runner link https://github.com/rook/rook/runs/8183513355?check_suite_focus=true

--- a/.github/workflows/build.yml
+++ b/.github/workflows/build.yml
@@ -34,24 +34,17 @@ jobs:
         with:
           version: v3.6.2
 
-      - name: copy working directory to GOPATH
-        run: sudo mkdir -p /Users/runner/go/src/github.com && sudo cp -a /Users/runner/work/rook /Users/runner/go/src/github.com/
-
       - name: build rook
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: |
           GOPATH=$(go env GOPATH) make clean && make -j$nproc IMAGES='ceph' BUILD_CONTAINER_IMAGE=false build
 
       - name: validate build
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh build
 
       - name: run codegen
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: GOPATH=$(go env GOPATH) make codegen
 
       - name: validate codegen
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh codegen
 
       - name: run mod check
@@ -61,19 +54,15 @@ jobs:
         run: tests/scripts/validate_modified_files.sh modcheck
 
       - name: run crds-gen
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: make csv-clean && GOPATH=$(go env GOPATH) make crds
 
       - name: validate crds-gen
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh crd
 
       - name: run gen-rbac
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: GOPATH=$(go env GOPATH) make gen-rbac
 
       - name: validate gen-rbac
-        working-directory: /Users/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh gen-rbac
 
       - name: consider debugging

--- a/.github/workflows/codegen.yml
+++ b/.github/workflows/codegen.yml
@@ -38,13 +38,8 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: copy working directory to GOPATH
-        run: sudo mkdir -p /home/runner/go/src/github.com && sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
-
       - name: run codegen
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: GOPATH=$(go env GOPATH) make codegen
 
       - name: validate codegen
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh codegen

--- a/.github/workflows/crds-gen.yml
+++ b/.github/workflows/crds-gen.yml
@@ -37,13 +37,8 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: copy working directory to GOPATH
-        run: sudo mkdir -p /home/runner/go/src/github.com && sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
-
       - name: run crds-gen
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: GOPATH=$(go env GOPATH) make crds
 
       - name: validate crds-gen
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh crd

--- a/.github/workflows/rbac-gen.yaml
+++ b/.github/workflows/rbac-gen.yaml
@@ -37,13 +37,8 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: copy working directory to GOPATH
-        run: sudo mkdir -p /home/runner/go/src/github.com && sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
-
       - name: run gen-rbac
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: GOPATH=$(go env GOPATH) make gen-rbac
 
       - name: validate gen-rbac
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: tests/scripts/validate_modified_files.sh gen-rbac

--- a/.github/workflows/unit-test.yml
+++ b/.github/workflows/unit-test.yml
@@ -38,9 +38,5 @@ jobs:
         with:
           go-version: 1.18
 
-      - name: copy working directory to GOPATH
-        run: sudo mkdir -p /home/runner/go/src/github.com && sudo cp -a /home/runner/work/rook /home/runner/go/src/github.com/
-
       - name: run unit tests
-        working-directory: /home/runner/go/src/github.com/rook/rook
         run: GOPATH=$(go env GOPATH) make -j $(nproc) test

